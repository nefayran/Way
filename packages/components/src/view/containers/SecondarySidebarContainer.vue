<template>
  <div class="wb-container">
    <div class="wb-sidebar ma-md pa-lg br-lg">
      <h2 v-if="wbComponent" class="text-accent--base font-secondary">
        {{ wbComponent.name }}
      </h2>
      <h2 v-else class="text-accent--base font-secondary">Component</h2>
      <h3 class="pt-lg">Properties:</h3>
      <ul
        v-if="wbComponent && properties"
        class="pl-no pt-lg white--base list-none"
      >
        <li
          v-for="(value, name, index) in properties"
          :key="index"
          class="text-md pb-md text-primary--accent-1"
        >
          <span>
            {{ name }}
          </span>
        </li>
      </ul>
    </div>
  </div>
</template>

<script async setup lang="ts">
import { computed, ref, watchEffect } from "vue";
import store from "@/app/store";

const wbComponent = computed(
  () => store.getters["ComponentsStore/getComponent"]
);
const properties = ref();
// const propsPromise = wbComponent.prop;
// const properties = ref(propsPromise());
// console.log(wbComponent.value.props);
// const properties = propsPromise.then((result: any) => result);
// const props = await wbComponent.value.props;

watchEffect(async () => {
  if (wbComponent.value && wbComponent.value.props) {
    properties.value = await wbComponent.value.props;
  }
});
// console.log(propsPromise));
</script>

<style lang="scss">
.wb-sidebar {
  height: calc(100% - var(--space-lg) * 3);
}
</style>
