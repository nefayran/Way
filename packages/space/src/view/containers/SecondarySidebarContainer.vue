<template>
  <div class="wb-container">
    <div class="wb-sidebar ma-md pa-lg br-lg">
      <h2 v-if="wbComponent" class="text-accent--base font-secondary">
        {{ wbComponent.name }}
      </h2>
      <h2 v-else class="text-accent--base font-secondary">Component</h2>
      <h3 class="pt-lg">Properties:</h3>
      <ul
        v-if="wbComponent && properties"
        class="pl-no pt-lg white--base list-none"
      >
        <li
          v-for="(value, name, index) in properties"
          :key="index"
          class="text-md pb-md text-primary--accent-1"
        >
          <span>
            {{ name }}
          </span>
        </li>
      </ul>
    </div>
  </div>
</template>

<script async setup lang="ts">
import { computed, ref, watchEffect } from "vue";
import store from "@/app/store";

const wbComponent = computed(
  () => store.getters["ComponentsStore/getComponent"]
);
const properties = ref();

watchEffect(async () => {
  if (wbComponent.value && wbComponent.value.props) {
    properties.value = await wbComponent.value.props;
  }
});
</script>

<style lang="scss">
.wb-sidebar {
  height: calc(100% - var(--space-lg) * 3);
}
</style>
