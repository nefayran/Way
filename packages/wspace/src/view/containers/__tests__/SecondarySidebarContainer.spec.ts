import { mount } from "@vue/test-utils";
import SecondarySidebarContainer from "../SecondarySidebarContainer.vue";

describe("SecondarySidebarContainer.vue", () => {
  test("should have an wb-container class", () => {
    const wrapper = mount(SecondarySidebarContainer, {});
    expect(wrapper.classes()).toContain("wb-container");
    expect(wrapper.element).toMatchSnapshot();
  });
  test("generate components list", () => {
    const wrapper = mount(SecondarySidebarContainer, {});
    expect(wrapper.classes()).toContain("wb-container");
    expect(wrapper.element).toMatchSnapshot();
  });
});
