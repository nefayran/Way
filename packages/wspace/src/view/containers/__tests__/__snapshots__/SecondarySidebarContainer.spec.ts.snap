// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondarySidebarContainer.vue generate components list 1`] = `
<div
  class="wb-container"
>
  <div
    class="wb-sidebar ma-md pa-lg br-lg"
  >
    <h2>
      -
    </h2>
    <h3
      class="pt-lg"
    >
      Properties:
    </h3>
    <!--v-if-->
  </div>
</div>
`;

exports[`SecondarySidebarContainer.vue should have an wb-container class 1`] = `
<div
  class="wb-container"
>
  <div
    class="wb-sidebar ma-md pa-lg br-lg"
  >
    <h2>
      -
    </h2>
    <h3
      class="pt-lg"
    >
      Properties:
    </h3>
    <!--v-if-->
  </div>
</div>
`;
